### PR TITLE
add webhook controller, schema, and persist events to slack

### PR DESCRIPTION
**Pull request by zach-bob**
This PR aims to implement [this Jira ticket](https://mojotech.atlassian.net/jira/software/projects/MRNN/boards/22?selectedIssue=MRNN-32)

### CODE REVIEW

**Comment by zach-bob** [comment]
There are tons of commits here because this PR is based from #22 and #22 is based from #21. At the time of writing this, these both have yet to be merged.

**Comment by zach-bob** [comment]
Re-adding DO NOT MERGE due to all of the dependent branches that have not been merged yet.

File: lib/release_notes_bot/repositories.ex [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  def get_all do
+    Repo.all(Repository)
+  end
+
+  def get(param) do
+    Repo.get_by(Repository, param)
+  end
+
+  def existence(raw_repository) do
```
Comment (sethkrasnianski) [inline lib/release_notes_bot/repositories.ex:23]: This sounds like a `get_or_create` function. Not sure `existence` really makes sense here as it's sort of implying it's a predicate and not something that produces side effects.

File: lib/release_notes_bot_web/controllers/slack_interaction_controller.ex [hunk 2]
```diff
@@ -4,34 +4,32 @@ defmodule ReleaseNotesBotWeb.SlackInteractionController do
...
   use ReleaseNotesBotWeb, :controller
-  alias ReleaseNotesBot.Projects
+  alias ReleaseNotesBot.{Projects, Channels}
 
   def index(conn, params) do
     %{"view" => view, "user" => user} = Projects.parse_params(params)
 
     case Projects.parse_response(view) do
       %{client: client_name, channel: channel} ->
-        Slack.Web.Chat.post_message(
```
Comment (sethkrasnianski) [inline lib/release_notes_bot_web/controllers/slack_interaction_controller.ex:14]: What happened to `Slack.Web.chat`? Is it still being used elsewhere? Can we remove it?

**Pull request merged**